adapter/oci/app_client.go

```diff
@@ -0,0 +1,208 @@
+package oci
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"github.com/fnproject/cli/adapter"
+	"github.com/go-openapi/strfmt"
+	"github.com/oracle/oci-go-sdk/functions"
+	"github.com/spf13/viper"
+	"os"
+	"time"
+)
+
+const (
+	// AnnotationSubnet - Subnet used to indicate the placement of the function runtime
+	AnnotationSubnet = "oracle.com/oci/subnetIds"
+)
+
+type AppClient struct {
+	client *functions.FunctionsManagementClient
+}
+
+func (a AppClient) CreateApp(app *adapter.App) (*adapter.App, error) {
+	compartmentId := viper.GetString("oracle.compartment-id")
+
+	subnetIds, err := parseSubnetIds(app.Annotations)
+
+	if err != nil {
+		return nil, err
+	}
+	body := functions.CreateApplicationDetails{
+		CompartmentId: &compartmentId,
+		Config:        app.Config,
+		DisplayName:   &app.Name,
+		SubnetIds:     subnetIds,
+	}
+	req := functions.CreateApplicationRequest{CreateApplicationDetails: body,}
+
+	res, err := a.client.CreateApplication(context.Background(), req)
```

Comment:
Would there be a benefit in taking a context as a parameter into these methods, rather than using `context.Background()`? (allow caller to set timeout, cancel etc.).

Reply:
We do not provide such facilities (set timeout, cancel etc.) to the caller in our current implementation that consumes Open Source Go CP APIs. Hence, let's keep it the same with the OCI APIs.